router/types/get_direct_custom_quote_request.go

```diff
@@ -0,0 +1,130 @@
+package types
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/osmosis-labs/sqs/domain"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/labstack/echo/v4"
+)
+
+// GetDirectCustomQuoteRequest represents
+type GetDirectCustomQuoteRequest struct {
+	TokenIn        *sdk.Coin
+	TokenOutDenom  []string
+	TokenOut       *sdk.Coin
+	TokenInDenom   []string
+	PoolID         []uint64 // list of the pool ID
+	ApplyExponents bool     // Boolean flag indicating whether to apply exponents to the spot price. False by default.
+}
+
+// UnmarshalHTTPRequest unmarshals the HTTP request to GetDirectCustomQuoteRequest.
+// It returns an error if the request is invalid.
+// NOTE: Currently method for some cases returns an error, while for others
+// it returns a response error. This is not consistent and should be fixed.
+func (r *GetDirectCustomQuoteRequest) UnmarshalHTTPRequest(c echo.Context) error {
```

Comment:
Not sure I'm following the consistency take here

Reply:
I am referring that within a method it returns `return c.JSON(domain.GetStatusCode(err), domain.ResponseError{Message: err.Error()})` or for example `return ErrTokenInNotValid`, which is inconsistent and sometimes lead to some confusion. I have ported such behaviour from original implementation to avoid different behaviour for the clients.

Reply:
I see - is there a reason why we cannot fix it right now by always returning an error and then letting the caller handle this as needed?

Reply:
I think it could be done right now, but at the same time I feel like we already spent some reasonable amount of time improving existing code base while working on this task. From my experience constantly iterating and improving bit by bit allows move faster and do not stuck too long on specific code blocks.

Reply:
No problems - I understand this stream of work has been ongoing for some time with many iterations. It should be pretty quick - I will do it in a follow-up PR